### PR TITLE
DashboardScene: Support remember scroll position when coming back from view panel, panel edit and settings

### PR DESCRIPTION
**Pull request by torkelo**
* Unifies handling of remember and restoring scroll position so they hare handled the same by consumers of NativeScrollbar no matter if bodyScrolling is enabled or not. Not only does this simplify / unify the code (less conditions), but could also be important even after bodyScrolling is made the only supported mode. If [sidecars](https://github.com/grafana/grafana/pull/91648) becomes a thing pages might need to switch to a div scroll container and for that we need some kind of abstractions that treats both modes the same. To fully support that we would need some kind of scroll context though to know when to switch to a div scroll container (not explored in this PR). 

* Remove scrollTop and handle setting scroll position the same way in both scenarios (via function call), not sure which is best here do it via prop in NativeScrollbar that then does a function call inside a useEffect. But since the new code in the old DashboardPage when bodyScrolling was enabled just called window.scrollTo I opted to follow this lead and simply do that. This reduces props and makes the interface a bit simpler. 

TODO 
* Likely need to update tests (unit & e2e) but will do that after an initial review

### CODE REVIEW

File: public/app/core/components/Page/Page.tsx [hunk 1]
```diff
@@ -57,9 +56,7 @@ export const Page: PageType = ({
         <NativeScrollbar
           // This id is used by the image renderer to scroll through the dashboard
           divId="page-scrollbar"
-          autoHeightMin={'100%'}
```
Comment (torkelo) [inline public/app/core/components/Page/Page.tsx:60]: this prop existed on CustomScrollbars (which NativeScrollbar used as Props type) but was never actually used so removing it here

File: public/app/core/components/Page/types.ts [hunk 2]
```diff
@@ -22,15 +24,11 @@ export interface PageProps extends HTMLAttributes<HTMLDivElement> {
...
    * @deprecated this will be removed when bodyScrolling is enabled by default
    * Can be used to get the scroll container element to access scroll position
    * */
-  scrollRef?: RefCallback<HTMLDivElement>;
-  /**
-   * @deprecated this will be removed when bodyScrolling is enabled by default
-   * Can be used to update the current scroll position
-   * */
-  scrollTop?: number;
+  onSetScrollRef?: (ref: ScrollRefElement) => void;
```
Comment (torkelo) [inline public/app/core/components/Page/types.ts:31]: only DashboardPage ever uysed scrollTop and scrollRef (checked enterprise as well), so safe to just remove / rename

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx [hunk 3]
```diff
@@ -925,6 +933,23 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
...
+  public rememberScrollPos() {
+    this._prevScrollPos = this._scrollRef?.scrollTop;
+    console.log('rememberScrollPos', this._prevScrollPos);
+  }
+
+  public restoreScrollPos() {
+    if (this._prevScrollPos !== undefined) {
+      setTimeout(() => {
+        this._scrollRef?.scrollTo(0, this._prevScrollPos!);
+      }, 50);
```
Comment (torkelo) [inline public/app/features/dashboard-scene/scene/DashboardScene.tsx:948]: this setTimeout is annoying but without it does not work, tried requestAnimationFrame but did work either.

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx [hunk 4]
```diff
@@ -925,6 +931,23 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
...
     });
   }
+
+  public onSetScrollRef = (scrollElement: ScrollRefElement): void => {
+    this._scrollRef = scrollElement;
+  };
+
+  public rememberScrollPos() {
+    this._prevScrollPos = this._scrollRef?.scrollTop;
+    console.log('rememberScrollPos', this._prevScrollPos);
```
Comment (mdvictor) [inline public/app/features/dashboard-scene/scene/DashboardScene.tsx:941]: Sneaky log

**Comment by mdvictor** [comment]
I tested this out and it looks and works great!

File: public/app/core/components/NativeScrollbar.tsx [hunk 5]
```diff
@@ -2,28 +2,35 @@ import { css, cx } from '@emotion/css';
...
+}
+
+export interface ScrollRefElement {
+  scrollTop: number;
+  scrollTo: (x: number, y: number) => void;
+}
 
 // Shim to provide API-compatibility for Page's scroll-related props
 // when bodyScrolling is enabled, this is a no-op
 // TODO remove this shim completely when bodyScrolling is enabled
```
Comment (ashharrison90) [inline public/app/core/components/NativeScrollbar.tsx:20]: i guess this isn't true anymore 😅 

what do we envisage this component looking like once `bodyScrolling` is enabled by default and all toggle code removed? 🤔 does this basically become the abstraction you're talking about in the description? would something like `PageScrollHelper` make sense as name in the future?

(don't have to do any of this now, just trying to get it clear in my head where this component is heading)

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx (hunk 4 above)
Comment (torkelo) [reply public/app/features/dashboard-scene/scene/DashboardScene.tsx:941]: the best logs are sneaky :)

File: public/app/core/components/NativeScrollbar.tsx (hunk 5 above)
Comment (torkelo) [reply public/app/core/components/NativeScrollbar.tsx:20]: yea, my guess is we will need some kind of abstraction (component) and a context if we are to support switching between body scrolling and div scrolling.

File: public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx [hunk 6]
```diff
@@ -26,6 +27,20 @@ export function DashboardSceneRenderer({ model }: SceneComponentProps<DashboardS
   const navModel = getNavModel(navIndex, 'dashboards/browse');
   const hasControls = controls?.hasControls();
 
+  // Remember scroll pos when going into view panel, edit panel or settings
+  useMemo(() => {
+    if (viewPanelScene || editview || editPanel) {
```
Comment (axelavargas) [inline public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx:32]: Here I found an edge case with `editview` (Settings), if we go to settings, and click on the variables tab(or other tab), the scroll position is updated to zero. 

https://github.com/user-attachments/assets/020bbcd2-7535-40dc-8580-94214f5bb257

Comment (torkelo) [reply public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx]: @axelavargas good catch, easy to fix

**Review by axelavargas — APPROVED** [review]
Gave it another test and works as expected 🥳

**Pull request merged**